lib/commands/promote.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const chalk = require('chalk');
+const isMissingData = require('./validations').isMissingData;
+const max_promotions = 3;
+
+let elapsedInterval;
+let spamSeed = generateSeed()
+
+let promoteCount = 0;
+function interrupt() {
+    return promoteCount++ >= max_promotions;
+}
+// WARNING: Not cryptographically secure. Do not use any seeds generated by this generator to actually store any value.
+function generateSeed() {
+    const validChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ9'
+    return Array.from(new Array(81), (x, i) => validChars[Math.floor(Math.random() * validChars.length)]).join('')
+}
+
+function generateTransfer() {
+    return {
+        address: spamSeed,
+        value: 0,
+        message: '',
+        tag: ''
+    }
+}
+const setupPromoteCommand = (data, iotajs, refreshAccountData, vorpal) => {
+  vorpal
+    .command('promote <hash>', 'Promotes a pending transaction.  Provide an id for additional details.')
+    .option('-n <number>', 'Max number of transactions.  Default 10.')
+    .autocomplete({
+      data: () => {
+        vorpal.ui.input(`promote ${vorpal.ui.input().split(' ')[1].toUpperCase()}`);
+        if (!data.accountData) {
+          return [];
+        }
+        return data.accountData.transfers.map(b => b[0]).map(t => t.hash.slice(0, 6));
+      }
+    })
+    .action((args, callback) => {
+      if (isMissingData(['node'])) {
+        return callback();
+      }
+
+      var hash = args.hash;
+      if (hash.length < 81) {
+        if (!data.accountData) {
+          vorpal.log(chalk.red('Please provide a full transaction hash'));
```

Comment:
I don't think this error message fits the situation.

Reply:
If a user has just connected and not entered a seed they can still promote so this was to catch those that may have tried `promote ABCDEF` when they've yet to enter a seed? Perhaps something like 'Promoting with a short hash is only possible with a history.' or something like that?

Reply:
Yeah, I prefer that.  Since the history downloads automatically it's probably just a matter of waiting.  In similar situations (like `balance`) I've said to try the command again in a few minutes once the transaction data has finished loading.